### PR TITLE
Support for decorated services

### LINKED ISSUE #20: Support for decorated services
The goal of this feature is to be able to override an existing definition, but keep having acces to the old one, as described here :
https://symfony.com/doc/current/service_container/service_decoration.html

I've started to work on a `decorate($entryId, $renameId, $priority)` method for the create definition helper.

Interested ?

### PR DESCRIPTION
**Pull request by pitchart**
Closes #20

### CODE REVIEW

File: tests/DecoratorTest.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        $decoratingClassName = get_class($decorating);
+
+        $container = new ContainerBuilder;
+        (new PhpConfigLoader($container))->load([
+            'decorated' => create($decoratedClassName),
+            'decorating' => create($decoratingClassName)
+                ->decorate('decorated')
+                ->arguments(get('decorating.inner')),
+        ]);
+        $this->process($container);
```
Comment (mnapoli) [inline tests/DecoratorTest.php:34]: How about calling `$container->compile()` instead? It should do the trick I think?

**Review by mnapoli — COMMENTED** [review]
👍  this is awesome, could you add a bit of documentation?

File: tests/DecoratorTest.php (hunk 1 above)
Comment (pitchart) [reply tests/DecoratorTest.php:34]: If we only call the `$container->compile()` method, we can not check if the decorated definition is available to use by other services, as the decorated service is marked as private, so not available form outside after compilation.

So we can remove assertions that test existance of the decorated service before compilation (we can consider that this is the dependency injection component's responsibility) or separate the tests, one for the `DecoratorServicePass` and another for the decorator classes.

Comment (mnapoli) [reply tests/DecoratorTest.php:34]: oh OK, then that's fine by me!

**Comment by mnapoli** [comment]
:+1: for merging, could you rebase or merge master? I tried to resolve conflicts in GitHub but I was messing up everything

**Comment by mnapoli** [comment]
Thanks!

**Pull request merged**